tests/unit_tests/test_simulator.py

```diff
@@ -0,0 +1,343 @@
+    }
+
+
+@pytest.fixture
+def corsikaFile():
+    return "run1_proton_za20deg_azm0deg-North-1LST_trigger_rates.corsika.zst"
+
+
+@pytest.fixture
+def array_simulator(set_db, label, arrayConfigData):
+
+    arraySimulator = Simulator(
+        label=label, simulator='simtel', configData=arrayConfigData)
+    return arraySimulator
+
+
+@pytest.fixture
+def shower_simulator(set_db, label, showerConfigData):
+
+    showerSimulator = Simulator(
+        label=label, simulator='corsika', configData=showerConfigData)
+    return showerSimulator
+
+
+def test_set_file_locations(array_simulator, shower_simulator):
+
+    assert Path.exists(array_simulator._simtelSourcePath)
+    assert Path.exists(array_simulator._filesLocation)
+    assert Path.exists(array_simulator._outputDirectory)
+
+    assert Path.exists(shower_simulator._simtelSourcePath)
+    assert Path.exists(shower_simulator._filesLocation)
+    assert Path.exists(shower_simulator._outputDirectory)
+
+
+def test_guess_run_from_file(array_simulator):
+
+    assert array_simulator._guessRunFromFile("run12345_bla_ble") == 12345
+
+    # Invalid run number - returns 1
+    assert array_simulator._guessRunFromFile("run1test2_bla_ble") == 1
+
+    assert array_simulator._guessRunFromFile("abc-run12345_bla_ble") == 12345
+
+    assert array_simulator._guessRunFromFile("run10") == 10
+
+    assert array_simulator._guessRunFromFile("abc-ran12345_bla_ble") == 1
+
+    # TODO add test for 'Error creating output directory'
+    # (not sure how to test a failed mkdir
+
+
+def test_set_simulator(array_simulator, shower_simulator):
+
+    array_simulator._set_simulator('simtel')
+    assert array_simulator.simulator == 'simtel'
+
+    shower_simulator._set_simulator('corsika')
+    assert shower_simulator.simulator == 'corsika'
+
+    with pytest.raises(gen.InvalidConfigData):
+        shower_simulator._set_simulator('this_simulator_is_not_there')
+
+
+def test_loadConfigData(array_simulator):
+
+    with pytest.raises(gen.InvalidConfigData):
+        array_simulator._loadConfigData()
+
+
+def test_loadCorsikaConfig(shower_simulator, showerConfigData):
+
+    shower_simulator._loadCorsikaConfig(configData=showerConfigData)
+
+    assert shower_simulator.site == "South"
+
+    assert "site" not in shower_simulator._corsikaConfigData
+
+    _temp_shower_data = copy(showerConfigData)
+    _temp_shower_data.pop("site")
+    with pytest.raises(KeyError):
+        shower_simulator._loadCorsikaConfig(configData=_temp_shower_data)
+
+
+def test_loadSimTelConfig(array_simulator, arrayConfigData):
+
+    array_simulator._loadSimTelConfig(arrayConfigData)
+
+    assert isinstance(array_simulator.arrayModel, ArrayModel)
+
+
+def test_validateRunListAndRange(shower_simulator):
+
+    assert not shower_simulator._validateRunListAndRange(None, None)
+
+    run_list = [1, 24, 3]
+
+    assert shower_simulator._validateRunListAndRange(
+        runList=run_list, runRange=None) == list(set(run_list))
```

Comment:
Why do you have to do `list(set(run_list))`?

Reply:
Same comments as in previous issue.